src/pcap/pcap.go

```diff
@@ -0,0 +1,115 @@
+// Package pcap provides types and interfaces to build a remote packet capturing
+// tool. For details about the different applications see the different packages
+// in `cmd/`.
+package pcap
+
+import (
+	"fmt"
+	"unicode"
+
+	"github.com/go-playground/validator/v10"
+	"go.uber.org/zap"
+)
+
+//go:generate protoc --go_out=. --go_opt=paths=source_relative --go-grpc_out=. --go-grpc_opt=paths=source_relative pcap.proto
+
+const (
+	// CompatibilityLevel indicates whether two parties are compatible. Once there is a change
+	// that requires both parties to be updated this value MUST be incremented by one. The calling
+	// party has to ensure that the compatibility level of the called party is equal or larger and
+	// refuse operation if it isn't.
+	CompatibilityLevel int64 = 0
```

Comment:
why not use semantic versioning? `1.0.0` is compatible with `1.1.1` but not compatible with `2.0.0`
seems more straightforward to me

Reply:
The goal semantic versioning is trying to achieve does overlap with what we need but there are two main reasons why we decided against semantic versioning:
- currently we only version the release as a whole, a breaking change in the release, however, doesn't necessarily mean a breaking change in the internal API
- this level should also be used in case of security issues, those would almost certainly not require a major version to be released but we still want to lock out outdated agents (which will only receive irregular updates if deployed as part of an application)

Reply:
I agree with Max. Also I think we can keep it simple without to have a reference to release version (semantic versioning)

Reply:
as discussed, we stick with `CompatibilityLevel` for now and keep further semantics out for now.